### PR TITLE
Add --data-collection-settings param to aks-preview extension to be consistent with az cli

### PR DESCRIPTION
**Pull request by ganga1980**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az aks create -g <clusterRgName> -n <clusterName>--enable-addons monitoring --enable-msi-auth-for-monitoring  --data-collection-settings datacollectionsettings.json

az aks enable-addons -a monitoring --enable-msi-auth-for-monitoring -g <clusterRg> -n <clusterName> --data-collection-settings datacollectionsettings.json

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by ganga1980** [comment]
Hi, @FumingZhang , Can you please review this PR? Btw, test failures related to the test_aks_create_and_update_with_http_proxy_config which are not related to my change. can you please look at why this test being failed? do we need to re-record test test_aks_create_and_update_with_http_proxy_config?

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/addonconfiguration.py
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/addonconfiguration.py:119]: Currently aks-preview relies on azure-cli of version [2.43.0](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/azext_aks_preview/azext_metadata.json), and that version of azure-cli does not yet contain [the change](https://github.com/Azure/azure-cli/pull/24777) (should be included in the upcoming [2.44.0](https://github.com/Azure/azure-cli/milestone/125) release) to be reused.

So either wait for azure-cli to complete the release and aks-preview to update the dependencies before making changes, or copy the code from the azure-cli side.

**Review by FumingZhang — CHANGES_REQUESTED** [review]

File: src/aks-preview/azext_aks_preview/addonconfiguration.py
Comment (ganga1980) [reply src/aks-preview/azext_aks_preview/addonconfiguration.py:119]: > Currently aks-preview relies on azure-cli of version [2.43.0](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/azext_aks_preview/azext_metadata.json), and that version of azure-cli does not yet contain [the change](https://github.com/Azure/azure-cli/pull/24777) (should be included in the upcoming [2.44.0](https://github.com/Azure/azure-cli/milestone/125) release) to be reused.
> 
> So either wait for azure-cli to complete the release and aks-preview to update the dependencies before making changes, or copy the code from the azure-cli side.

> Sure, will wait for the aks-preview to get the updated for 2.44.0.  The official release date of 2.44.0 is 1/10/23, what's the ETA for aks-preview to get updated for this version?

Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/addonconfiguration.py:119]: I'll bump the dependency soon after the release of azure-cli, eta 01/13.

Comment (ganga1980) [reply src/aks-preview/azext_aks_preview/addonconfiguration.py:119]: Hi, @FumingZhang  - can you please approve and merge this PR after bumping the dependency?

Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/addonconfiguration.py:119]: Hey @ganga1980, opened this PR [#5759](https://github.com/Azure/azure-cli-extensions/pull/5759) to bump the dependency.

Comment (ganga1980) [reply src/aks-preview/azext_aks_preview/addonconfiguration.py:119]: > Hey @ganga1980, opened this PR [#5759](https://github.com/Azure/azure-cli-extensions/pull/5759) to bump the dependency.

> @FumingZhang ,  I see this PR https://github.com/Azure/azure-cli-extensions/pull/5759 already. can you please review and merge this PR as well?

Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/addonconfiguration.py:119]: Well, there are some conflicts in your PR. Also it seems something is missing when syncing the change to aks-preview. I've opened this new [PR #5779](https://github.com/Azure/azure-cli-extensions/pull/5779) based on your changes. PTAL

Comment (ganga1980) [reply src/aks-preview/azext_aks_preview/addonconfiguration.py:119]: Hi, @FumingZhang - Thanks for  resolving conflicts and making PR. Changes in the PR https://github.com/Azure/azure-cli-extensions/pull/5779, looks good to me. Once you merged that PR, please close this PR.

Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/addonconfiguration.py:119]: Sure

**Comment by FumingZhang** [comment]
Replaced by PR [#5779](https://github.com/Azure/azure-cli-extensions/pull/5779).

**Pull request closed without merging**
